src/dynamics/ads_player/ads_player.js

```diff
@@ -343,6 +347,21 @@ Scoped.define("module:Ads.Dynamics.Player", [
                         this._videoElement = this.parent() && this.parent().activeElement().querySelector("[data-video='video']"); // TODO video element for outstream
                     return this._videoElement;
                 },
+                renderVideoFrame: function(mediaUrl, width, height) {
+                    const video = document.createElement("video");
```

Comment:
Hi @maxblessed, can you use the existing video element instead of creating a new one? This will redownload the video and screenshot the first frame.